src/west/app/project.py

```diff
@@ -1501,12 +1512,56 @@ def ensure_cloned(self, project, stats, take_stats):
             if take_stats:
                 stats['init'] = perf_counter() - start
 
+    def create_auto_cache_info(self, project, cache_dir):
+        # auto-cache helper: Create some short informal file with basic info
+        # about the local cache folder.
+        content = textwrap.dedent(f"""
+        The following local cache directory  was automatically created by west:
+        - Local Cache:  {basename(cache_dir)}
+
+        It caches the following project:
+        - Project Name: {project.name}
+        - Project Path: {project.path}
+        - Project Url:  {project.url}
+        """)
+        with open(cache_dir + '.info', 'w') as f:
+            f.write(content)
+
+    def handle_auto_cache(self, project):
+        # update() helper. Initialize the specified cache directory if it has
+        # not been cloned yet. If the cache directory is already existing, it
+        # will be synced with remote.
+
+        cache_dir = self.project_cache(project)
+        auto_cache_dir = self.project_auto_cache(project)
+
+        if not auto_cache_dir or (cache_dir != auto_cache_dir):
+            # nothing to do: auto-cache directory was either not specified
+            # or another cache was specified with higher priority.
+            return
```

Comment:
We can do a slight improvement here:

```python
        auto_cache_dir = self.project_auto_cache(project)

        if not auto_cache_dir:
            # nothing to do: auto-cache directory was not specified
            return
            
        cache_dir = self.project_cache(project)
        if cache_dir != auto_cache_dir)
            # another cache was specified with higher priority.
            return
```

We can then replace
```python
        if self.auto_cache:
            self.handle_auto_cache(project)
```
with simply (as it will be a no-op)
```python
         self.handle_auto_cache(project)
```